ts-relayer-tests/src/ics721.spec.ts

```diff
@@ -311,95 +391,366 @@ test.serial(
+    )}`
+  );
+
+  // assert NFT minted on chain B
+  osmoClassId = `${onlyOsmoIncomingChannel.channel.dest.portId}/${onlyOsmoIncomingChannel.channel.dest.channelId}/${t.context.wasmCw721}`;
+  osmoCw721 = await osmoClient.sign.queryContractSmart(osmoIcs721, {
+    nft_contract: { class_id: osmoClassId },
+  });
+  allNFTs = await allTokens(osmoClient, osmoCw721);
+  t.true(allNFTs.tokens.length === 1);
+  // assert NFT on chain B is owned by osmoAddr
+  tokenOwner = await ownerOf(osmoClient, osmoCw721, tokenId);
+  t.is(osmoAddr, tokenOwner.owner);
+  // assert NFT on chain A is locked/owned by ICS contract
+  tokenOwner = await ownerOf(wasmClient, wasmCw721, tokenId);
+  t.is(wasmIcs721, tokenOwner.owner);
+  // assert NFT on chain B is owned by osmoAddr
+  osmoClassId = `${onlyOsmoIncomingChannel.channel.dest.portId}/${onlyOsmoIncomingChannel.channel.dest.channelId}/${t.context.wasmCw721}`;
+  osmoCw721 = await osmoClient.sign.queryContractSmart(osmoIcs721, {
+    nft_contract: { class_id: osmoClassId },
+  });
+  tokenOwner = await ownerOf(osmoClient, osmoCw721, tokenId);
+  t.is(osmoAddr, tokenOwner.owner);
+
+  // test back transfer NFT to wasm chain, where onlyOsmoIncomingChannel is not WLed on wasm chain
+  t.log(
+    `transfering back to wasm chain via unknown ${onlyOsmoIncomingChannel.channel.dest.channelId}`
+  );
+  transferResponse = await sendNft(
+    osmoClient,
+    osmoCw721,
+    t.context.osmoCw721OutgoingProxy,
+    {
       receiver: wasmAddr,
-      channel_id: otherChannel.channel.dest.channelId,
+      channel_id: onlyOsmoIncomingChannel.channel.dest.channelId,
       timeout: {
         block: {
           revision: 1,
           height: 90000,
         },
       },
-    };
-
-    t.log(
-      `transfering to wasm chain via ${otherChannel.channel.dest.channelId}`
-    );
+    },
+    tokenId
+  );
+  t.truthy(transferResponse);
+  // before relay NFT escrowed by ICS721
+  tokenOwner = await ownerOf(osmoClient, osmoCw721, tokenId);
+  t.is(osmoIcs721, tokenOwner.owner);
 
-    transferResponse = await sendNft(
-      osmoClient,
-      osmoCw721,
-      osmoCw721OutgoingProxy,
-      ibcMsg,
-      tokenId
-    );
-    t.truthy(transferResponse);
+  // Relay and verify we got an error
+  t.log("relaying packets");
+  info = await onlyOsmoIncomingChannel.link.relayAll();
+  for (const ack of info.acksFromB) {
+    const parsed = JSON.parse(fromUtf8(ack.acknowledgement));
+    t.log(`- ack: ${JSON.stringify(parsed)}`);
+  }
+  assertAckErrors(info.acksFromB);
 
-    t.log("relaying packets");
+  // assert after failed relay, NFT on chain B is returned to owner
+  allNFTs = await allTokens(osmoClient, osmoCw721);
+  t.true(allNFTs.tokens.length === 1);
+  // assert NFT is returned to sender on osmo chain
+  tokenOwner = await ownerOf(osmoClient, osmoCw721, tokenId);
+  t.is(osmoAddr, tokenOwner.owner);
 
-    info = await otherChannel.link.relayAll();
+  // ==== WL channel on wasm chain and test back transfer again ====
```

Comment:
migrate and WL channel for testing succesful back transfer, now NFT is unescrowd on existing collection - and NOT minted on a new collection